week-10-202402/Headers/Exercise1/AnimacionFigurasMov.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+#include "Controladora.h"
+
+namespace Ejercicio1Semana10 {
+
+	using namespace System;
+	using namespace System::ComponentModel;
+	using namespace System::Collections;
+	using namespace System::Windows::Forms;
+	using namespace System::Data;
+	using namespace System::Drawing;
+
+	/// <summary>
+	/// Resumen de MyForm
+	/// </summary>
+	public ref class MyForm : public System::Windows::Forms::Form
+	{
+	public:
+		MyForm(void)
+		{
+			InitializeComponent();
+			controladora = gcnew Controladora();
+		}
+
+	protected:
+		/// <summary>
+		/// Limpiar los recursos que se est�n usando.
+		/// </summary>
+		~MyForm()
+		{
+			if (components)
+			{
+				delete components;
+			}
+		}
+
+	private:
+		Controladora^ controladora;
+	private: System::Windows::Forms::Timer^ timer1;
+	private: System::ComponentModel::IContainer^ components;
+
+
+#pragma region Windows Form Designer generated code
+		/// <summary>
+		/// M�todo necesario para admitir el Dise�ador. No se puede modificar
+		/// el contenido de este m�todo con el editor de c�digo.
+		/// </summary>
+		void InitializeComponent(void)
+		{
+			this->components = (gcnew System::ComponentModel::Container());
+			this->timer1 = (gcnew System::Windows::Forms::Timer(this->components));
+			this->SuspendLayout();
+			// 
+			// timer1
+			// 
+			this->timer1->Enabled = true;
+			this->timer1->Tick += gcnew System::EventHandler(this, &MyForm::timer1_Tick);
+			// 
+			// MyForm
+			// 
+			this->AutoScaleDimensions = System::Drawing::SizeF(6, 13);
+			this->AutoScaleMode = System::Windows::Forms::AutoScaleMode::Font;
+			this->ClientSize = System::Drawing::Size(995, 528);
+			this->Name = L"MyForm";
+			this->Text = L"MyForm";
+			this->Load += gcnew System::EventHandler(this, &MyForm::MyForm_Load);
+			this->KeyDown += gcnew System::Windows::Forms::KeyEventHandler(this, &MyForm::MyForm_KeyDown);
+			this->ResumeLayout(false);
+
+		}
+#pragma endregion
+	private: System::Void MyForm_Load(System::Object^ sender, System::EventArgs^ e) {
+	}
+	private: System::Void timer1_Tick(System::Object^ sender, System::EventArgs^ e) {
+		Graphics^ g = this->CreateGraphics();
+		BufferedGraphicsContext^ espacio = BufferedGraphicsManager::Current;
+		BufferedGraphics^ buffer = espacio->Allocate(g, this->ClientRectangle);
+		buffer->Graphics->Clear(Color::White);
+		controladora->MoverFiguras(buffer->Graphics);
+		controladora->DibujarFiguras(buffer->Graphics);
+		buffer->Render(g);
+		delete buffer, espacio, g;
+	}
+	private: System::Void MyForm_KeyDown(System::Object^ sender, System::Windows::Forms::KeyEventArgs^ e) {
+		char keyPressed = toupper(e->KeyValue);
```

Comment:
Just as a recommendation we can work directly with the `e->KeyValue`
---
And switch cases will look like `Keys::T`, `Keys::C`, etc.